plugins/inputs/mqtt_consumer/mqtt_consumer.go

```diff
@@ -240,6 +245,36 @@ func compareTopics(expected []string, incoming []string) bool {
 }
 
 func (m *MQTTConsumer) onMessage(acc telegraf.TrackingAccumulator, msg mqtt.Message) error {
+	var remainingLength int
+	var qosFlagsSize int
+	var qOsSize int
+	topicSize := len(msg.Topic()) + 2
+	switch int(msg.Qos()) {
+	case 1:
+		qOsSize = 2
+		qosFlagsSize = 4
+	case 2:
+		qOsSize = 2
+		qosFlagsSize = 12
+	default:
+		qOsSize = 0
+	}
+	payloadSize := len(msg.Payload())
+	remainingContent := topicSize + qOsSize + payloadSize
+	if remainingContent <= 127 {
+		remainingLength = 1
+	} else if remainingContent <= 16383 {
+		remainingLength = 2
+	} else if remainingContent <= 2097151 {
+		remainingLength = 3
+	} else {
+		remainingLength = 4
+	}
```

Comment:
Just out of interest, wouldn't this be easier to get
```suggestion
	if remainingContent < 1 << 7 {
		remainingLength = 1
	} else if remainingContent < 1 << 14 {
		remainingLength = 2
	} else if remainingContent < 1 << 21 {
		remainingLength = 3
	} else {
		remainingLength = 4
	}
```

Reply:
Yes that would work as well. Im still new to Go so I wouldn't have thought of that at first, but looking at what it does, yes I'll go ahead and commit that in, thank you!